tmt/steps/report/junit.py

```diff
@@ -305,11 +305,24 @@ def _read_log_filter(log: Path) -> str:
         except tmt.utils.FileError:
             return ''
 
+    def _read_failures_filter(paths: list[Path]) -> str:
+        """
+        Read the contents of a given failures file
+        """
+        failures: list[str] = []
+        for path in paths:
+            try:
+                failures.extend(tmt.utils.yaml_to_list(phase.step.plan.execute.read(path)))
+            except tmt.utils.FileError:
+                continue
+
+        return '\n'.join(failures)
+
     environment.filters.update(
         {
             'read_log': _read_log_filter,
             'duration_to_seconds': _duration_to_seconds_filter,
-            'failures': tmt.result.Result.failures,
+            'read_failures': _read_failures_filter,
```

Comment:
I would keep the filter name, `failures` returning a list of failures looks good to me.

Reply:
Plus, if you use my suggestion to use `failure_logs` for the property returning paths, you can keep `failures` as a result property, iterating over its own `failure_logs`, and yielding actual failures. And we wouldn't need this custom filter at all, because it ould move into results, and other code could also use to "iterate of result failures". The piece in the template would be something like `result.failures | join("\n") | e`.

Reply:
I see, I missed that. Ok, let's stick to `failure_logs | failures`.